### ISSUE TITLE
Use dotnet-install.ps1 in Windows Dockerfiles

### ISSUE THREAD

**Issue by naamunds** [dotnet/dotnet-docker#88]
The Windows Server Dockerfiles should use [dotnet-install.ps1 from CLI](https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/obtain/dotnet-install.ps1) to install the CLI tools rather than directly containing the installation logic.  This would decouple the installation logic and also simplify the Dockerfiles.

This is currently blocked by https://github.com/dotnet/cli/issues/3113.

**Comment by MichaelSimons** [comment]
I recommend against this for a couple reasons.  First the dotnet-install.ps1 is not a cli build artifact.  Because of this, there won't be a SHA available to perform a checksum validation against when downloaded.  Second, the script does not handle adding dotnet to the path very well.  Within the Docker image, dotnet should be installed to Program Files and be available to all users of the image.  The script does not have support for modifying the machine path.  Additionally, the scripts path logic doesn't work when Powershell is launched as a child process as is the case in the Dockerfile - https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/obtain/dotnet-install.ps1#L391.

The bottom line is installing dotnet is pretty simple - download dotnet zip, extract, add to path.  Using the script at this point does not make sense for the Docker scenario.  If something changes in the future a new issue should be logged for reconsideration - closing issue.
